openpype/hosts/fusion/plugins/publish/collect_comp_frame_range.py

```diff
@@ -1,43 +1,98 @@
-        start, end, global_start, global_end = get_comp_render_range(comp)
-        context.data["frameStart"] = int(start)
-        context.data["frameEnd"] = int(end)
-        context.data["frameStartHandle"] = int(global_start)
-        context.data["frameEndHandle"] = int(global_end)
-        context.data["handleStart"] = int(start) - int(global_start)
-        context.data["handleEnd"] = int(global_end) - int(end)
+        # Get attributes
+        attributeValues = self.get_attr_values_from_data(context.data)
+        if attributeValues.get("frame_range_type") == "asset_render":
+            asset_doc = get_current_project_asset()
+            asset_start = asset_doc["data"]["frameStart"]
+            asset_end = asset_doc["data"]["frameEnd"]
+            asset_handle_start = asset_doc["data"]["handleStart"]
+            asset_handle_end = asset_doc["data"]["handleEnd"]
+
+            # Convert any potential none type to zero
+            asset_handle_start = asset_handle_start or 0
+            asset_handle_end = asset_handle_end or 0
+
+            # Calcualte in/out points
+            asset_global_start = asset_start - asset_handle_start
+            asset_global_end = asset_end + asset_handle_end
+
+            # If the validation is off, set the frameStart/EndHandle to the
+            # current frame range, so that's what will be rendered later on
+            context.data["frameStart"] = int(asset_start)
+            context.data["frameEnd"] = int(asset_end)
+            context.data["frameStartHandle"] = int(asset_global_start)
+            context.data["frameEndHandle"] = int(asset_global_end)
+            context.data["handleStart"] = int(asset_handle_start)
+            context.data["handleEnd"] = int(asset_handle_end)
+            context.data["frame_range_type"] = "asset_render"
+        else:
+            # Get the comps range
+            comp = context.data["currentComp"]
+            (
+                comp_start,
+                comp_end,
+                comp_global_start,
+                comp_global_end,
+            ) = get_comp_render_range(comp)
+
+            context.data["frameStart"] = int(comp_start)
+            context.data["frameEnd"] = int(comp_end)
+            context.data["frameStartHandle"] = int(comp_start)
+            context.data["frameEndHandle"] = int(comp_end)
+            context.data["handleStart"] = 0
+            context.data["handleEnd"] = 0
+            context.data["frame_range_type"] = "freerange_render"
+
+        self.log.info(
+            'Setting "frame_range_type" to "{}"'.format(
+                context.data["frame_range_type"]
+            )
+        )
+
+    @classmethod
+    def get_attribute_defs(self):
+        frame_ranges = {
+            "asset_render": "Asset's frame range",
+            "freerange_render": "Fusion's current frame range",
+        }
+
+        return [
+            EnumDef(
+                "frame_range_type",
+                items=frame_ranges,
```

Comment:
Maybe we could have the `dict` declaration in here directly, instead  of on line `78` ?